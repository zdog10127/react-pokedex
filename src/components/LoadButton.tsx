import React from "react";
import { ReactComponent as PokeballIcon } from "../svg/pokeball.svg";

type Props = {
  clickHandler: () => void;
};

const LoadButton = ({ clickHandler }: Props) => {
  return (
    <button
      onClick={clickHandler}
      className="py-2 px-6 rounded-lg font-semibold bg-primary text-white relative inline-flex focus:outline-none transform hover:-translate-y-1 transition-all ease-in-out duration-200"
    >
      <div className="absolute inset-y-0 left-0 flex items-center pl-4">
        <PokeballIcon className="w-6 h-6 fill-current" />
      </div>
      <span className="pl-8">Load More</span>
    </button>
  );
};

export default LoadButton;
