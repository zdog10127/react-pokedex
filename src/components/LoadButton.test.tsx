import React from "react";
import { render } from "@testing-library/react";
import LoadButton from "./LoadButton";

it("renders LoadButton successfully", () => {
  const { container, getByText } = render(
    <LoadButton clickHandler={() => {}} />
  );

  expect(getByText("Load More")).toBeTruthy();
  expect(container).toMatchSnapshot();
});
