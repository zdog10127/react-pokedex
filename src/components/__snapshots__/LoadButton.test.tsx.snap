// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoadButton successfully 1`] = `
<div>
  <button
    class="py-2 px-6 rounded-lg font-semibold bg-primary text-white relative inline-flex focus:outline-none transform hover:-translate-y-1 transition-all ease-in-out duration-200"
  >
    <div
      class="absolute inset-y-0 left-0 flex items-center pl-4"
    >
      <svg
        class="w-6 h-6 fill-current"
      >
        pokeball.svg
      </svg>
    </div>
    <span
      class="pl-8"
    >
      Load More
    </span>
  </button>
</div>
`;
